### DISCUSSION TITLE
Magento2 Driver with pub as Doc root

### DISCUSSION THREAD

**Discussion by davidverholen** [laravel/valet#1119] [Ideas]
currently, Magento 2 does not seem to work in developer mode with the pub directory as Document Root (which is best practice and should be encouraged).

I tried valet today for the first time and it seems to be really great, but I had to change the magento 2 driver (or I was too stupid and missing something) to get it working with the pub directory as doc root.

For us not using the pub dir as doc root is not an alternative (and should not be forced for anyone tbh.)

So I'm just wondering if that maybe already is possible, and I'm just a valet noob, or if that is an issue that should be addressed.

Since it seems quite easy to implement a valet driver, I would also be willing to look into this if you think it makes sense, too.

In Addition to this, maybe some configuration to set the website / store view per valet project would be a nice feature if this can be added in the driver.

**Comment by joaolisboa** [comment]
I had the same issue. The default driver doesn't seem to work, at least in developer mode it didn't for me. I switched to this driver - https://gist.github.com/gpressutto5/ee645498a0a66116c9bac3f42212d21a - which is working fine so far. All static and media is actually working now. Hoping this is looked into in a future version, because Magento 2 support is not great out-of-the-box from my experience.

**Comment by mattstauffer** [comment]
I'd welcome a pull request fixing it, but I have no experience with Magento and no test sites, so it would take multiple Magento-using folks (and possibly even someone from the Magento core team, if possible) working together to help it get fixed
